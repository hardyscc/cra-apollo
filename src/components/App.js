import React, { Component } from 'react'
import { Query } from 'react-apollo'
import { gql } from 'apollo-boost'

export const STOCKS_QUERY = gql`
  query {
    stocks(type: EQTY, first: 10, skip: 0) {
      code
      name
    }
  }
`

export const AppComponent = ({ loading, error, data }) => {
  if (loading) return <div id="loading">Loading...</div>
  if (error) return <div id="error">Error :(</div>
  return (
    <div id="success">
      Stocks :
      <ul>
        {data.stocks.map(stock => (
          <li key={stock.code}>
            {stock.code} {stock.name}
          </li>
        ))}
      </ul>
    </div>
  )
}

export const App = () => (
  <Query query={STOCKS_QUERY}>{result => <AppComponent {...result} />}</Query>
)
