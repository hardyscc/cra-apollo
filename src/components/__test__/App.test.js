import React from 'react'
import { mount } from 'enzyme'
import { MockedProvider } from 'react-apollo/test-utils'
import gql from 'graphql-tag'
import { graphql } from 'react-apollo'
import { STOCKS_QUERY, App } from '../App'
import { wait } from 'dom-testing-library'

it('renders welcome message', async () => {
  const data = { stocks: [{ code: '0001.HK', name: '0001' }] }
  const mocks = [
    {
      request: { query: STOCKS_QUERY },
      result: { data }
    }
  ]
  const tree = mount(
    <MockedProvider mocks={mocks}>
      <App />
    </MockedProvider>
  )

  await wait(() => {
    tree.update()
    return expect(tree.find('#stocks')).toHaveLength(1)
  })

  expect(tree.find('#stocks')).toMatchSnapshot()
})
