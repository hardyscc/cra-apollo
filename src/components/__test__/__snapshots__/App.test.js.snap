// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders welcome message 1`] = `
<div
  id="stocks"
>
  Stocks :
  <ul>
    <li
      key="0001.HK"
    >
      0001.HK
       
      0001
    </li>
  </ul>
</div>
`;
